### PR TITLE
#175 Feature - explicit charset option

### LINKED ISSUE #175: (Request) Easily specify EBCDIC codepage to use
Currently the tns encoding defaults to 'cp1047' which supports the Latin-1 character set.
This can result in the terminal showing some characters incorrectly if the mainframe uses a different codepage.  
For example, a mainframe using cp277 (Danish/Norwegian) will have the letter Ø displayed as @.

I did not find an obvious way to pass a specific codepage to the Emulator - or have I missed something?

A workaround is to explicitly instantiate ati, set the codepage and pass this to the Emulator:
```python 
from tnz.py3270 import Emulator
from tnz.ati import ati
ati.set("SESSION_CODE_PAGE", "cp277")
em = Emulator(
    visible=True,
    args=[
        "-trace",
        "-tracefile",
        "tnz_example.log",
    ],
    ati=ati
)
```

It would be nicer to be able to pass the codepage as an argument to the Emulator constructor, e.g.
```python 
from tnz.py3270 import Emulator
em = Emulator(
    visible=True,
    args=[
        "-trace",
        "-tracefile",
        "tnz_example.log",
        "-session_code_page", 
        "cp277"
    ]
)
```

I'm happy to put together a PR for this if you'd like and would appreciate input on naming the argument/option.

### PR DESCRIPTION
**Pull request by paullilley**
This PR resolves #175.

### CODE REVIEW

**Review by najohnsn — COMMENTED** [review]
can you ditch the dotenv stuff? it seems like additional complication that is not necessary to demonstrate the function

**Comment by paullilley** [comment]
Done :-)

**Comment by v1gnesh** [comment]
qq - will setting the env SESSION_CODE_PAGE with a possible option allow zti to use it as well?

**Comment by najohnsn** [comment]
Any SESSION_ environmental variable (including SESSION_CODE_PAGE) is picked up by tnz.ati/zti.

**Comment by paullilley** [comment]
Nice :-) 
@najohnsn do you want any further changes on this PR (I've removed the dotenv stuff and merged the lastet from main) ?
There's 1 workflow awaiting approval, but I guess you see that.

**Review by najohnsn — COMMENTED** [review]
Also, could you please update the `py3270.py` `Copyright 2021` to `Copyright 2021, 2024`?

File: tnz/py3270.py [hunk 1]
```diff
@@ -989,6 +1028,13 @@ def __init__(self, visible=False, timeout=30,
             idx = args.index("-tracefile")
             if idx >= 0:
                 ati.set("LOGDEST", args[idx+1])
+            idx = args.index("-charset")
+            if idx >= 0:
+                try:
+                    code_page = CHARSET_CODE_PAGES[args[idx+1]]
+                    ati.set("SESSION_CODE_PAGE", code_page)  
+                except (KeyError, IndexError):
+                    pass  # fail silently?
```
[the remark this answers is not in the document]
Comment (paullilley) [reply tnz/py3270.py]: Done

File: examples/logon_x.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
...
 em = Emulator(visible=True, args=["-trace",
                                   "-tracefile",
-                                  "example.log"])
-em.connect('mvs1')
-
+                                  "example.log",
+                                  "-charset", 
+                                  "cp1047",
+                                  ])
+em.connect(host="mvs1", port="992")
```
[the remark this answers is not in the document]
Comment (paullilley) [reply examples/logon_x.py]: Done

File: requirements.txt [hunk 3]
```diff
@@ -10,4 +10,4 @@ pylint==3.3.1
 pytest==8.3.3; python_version>="3.8"
 pytest>=7.4.2; python_version<"3.8"
 twine==5.1.1; sys_platform!="zos"
-wheel==0.45.0
+wheel==0.45.0
```
[the remark this answers is not in the document]
Comment (paullilley) [reply requirements.txt]: Fixed

File: .gitignore [hunk 4]
```diff
@@ -11,3 +11,12 @@ _version.py
...
 *.sw?
+
+# virtual environment
+.venv
+
+# VScode editor configs
+.vscode/
+
+# specific environment variables, e.g. host, user, password
+.env
```
[the remark this answers is not in the document]
Comment (paullilley) [reply .gitignore]: Fixed

**Comment by najohnsn** [comment]
@paullilley, looks like there are some pycodestyle things to address

**Comment by paullilley** [comment]
@najohnsn pycodestyle fixes done :-)

**Review by najohnsn — APPROVED** [review]
Looks great! Thanks for your contribution and for bearing with me!

**Pull request merged**